src/UI/CustomInput/CustomInput.tsx

```diff
@@ -0,0 +1,58 @@
+import { ChangeEventHandler, useEffect, useState } from 'react';
+import './CustomInput.scss';
+
+interface ICustomInputProps {
+  type?: 'text' | 'email' | 'password' | 'number';
```

Comment:
For this variant of props better if we create object with data about types. For example:  
```
export const INPUT_TYPES = {
  TEXT: 'text',
  EMAIL: 'email',
  PASSWORD: 'password',
  NUMBER: 'number',
} as const;

export type InputType = typeof InputTypes[keyof typeof InputTypes];

//usage example:

type = InputTypes.TEXT
```

Reply:
I agree, but maybe rework it's later.